models.py

```diff
@@ -161,17 +175,37 @@ def forward(self,
     previous_embedding = self.output_embedding(input)
     # Compute context vector.
     h = previous_state[0]
-    context_vector = self.additive_attention(h, encoder_states, attention_mask)
+    context_vector, attention = self.additive_attention(h, encoder_states, attention_mask)
     # Compute lstm input.
     lstm_input =  torch.cat((previous_embedding, context_vector), dim=-1)
+    if self.training:
+      # Add dropout to lstm input.
+      dropout = nn.Dropout(p=self.drop_out)
+      lstm_input = dropout(lstm_input)
     # Compute new decoder state.
     decoder_state = self.lstm_cell(lstm_input, previous_state)
     # Compute classifier input (alternatively a pre-output can be computed
     # before with a NN layer).
+
     classifier_input = torch.cat(
       (previous_embedding, decoder_state[0], context_vector), dim=-1)
     predictions = self.classifier(classifier_input)
-    return decoder_state, predictions
+
+    # generation probability
+    if self.use_pointer_generator:
+      p_gen_input = torch.cat((context_vector, decoder_state[0], lstm_input), 1)
```

Comment:
who is dimension 1? is it the same with dim=-1? if yes, maybe that would be preferable